### PR TITLE
Implement a page with the information about a change

### PR DESCRIPTION
**Pull request by flepied**
- db: repository_fullname_and_number -> change_id
- db: do not use / and # in change_id to be url friendly
- web: implement /:index/change/:change_id

Issue: https://github.com/morucci/monocle/issues/73

### CODE REVIEW

File: web/src/App.js [hunk 1]
```diff
@@ -201,13 +198,59 @@ OpenChangesView.propTypes = {
...
+  render () {
+    console.log(this.props.match.params.change)
+    return (
+      <React.Fragment>
+        <TopMenu index={this.props.match.params.index}/>
+        <Container>
+          <Row><Col><p></p></Col></Row>
+          <Row>
+            <Col>
+              <CFiltersForm index={this.props.match.params.index}/>
```
Comment (morucci) [inline web/src/App.js:211]: Why to keep the filterbox ?

File: monocle/db/queries.py [hunk 2]
```diff
@@ -619,3 +610,26 @@ def oldest_open_changes(es, index, repository_fullname, params):
     data = run_query(es, index, body)
     changes = [r['_source'] for r in data['hits']['hits']]
     return {'items': changes, 'total': data['hits']['total']}
+
+
+def all_changes(es, index, repository_fullname, params):
```
Comment (morucci) [inline monocle/db/queries.py:615]: I would have preferred "changes_and_events" ?

File: monocle/db/queries.py [hunk 3]
```diff
@@ -90,6 +85,8 @@ def generate_filter(repository_fullname, params):
         qfilter.append({"terms": {"author": authors}})
     if on_authors:
         qfilter.append({"terms": {"on_author": on_authors}})
+    if change_ids:
+        qfilter.append({"terms": {"change_id": change_ids}})
     if 'Change' in params['etype']:
```
Comment (morucci) [inline monocle/db/queries.py:90]: You should change this to:
if 'Change' in params['etype'] and len(params['etype"]) == 1:
to not enter in this condition for the new all_changes function

File: monocle/db/queries.py [hunk 4]
```diff
@@ -56,17 +56,11 @@ def generate_events_filter(params, qfilter):
...
         qfilter.append({"term": {"state": state}})
-    if repository_fullname_and_number:
-        qfilter.append(
-            {
-                'terms': {
-                    "repository_fullname_and_number": repository_fullname_and_number
-                }
-            }
-        )
+    if change_id:
```
Comment (morucci) [inline monocle/db/queries.py:62]: Then you should remove this

File: monocle/db/queries.py [hunk 5]
```diff
@@ -56,17 +56,11 @@ def generate_events_filter(params, qfilter):
 
 def generate_changes_filter(params, qfilter):
     state = params.get('state')
-    repository_fullname_and_number = params.get('repository_fullname_and_number')
+    change_id = params.get('change_id')
```
Comment (morucci) [inline monocle/db/queries.py:59]: Then you should remove this

File: monocle/db/queries.py [hunk 6]
```diff
@@ -77,6 +71,7 @@ def generate_filter(repository_fullname, params):
     on_authors = params.get('on_authors')
     exclude_authors = params.get('exclude_authors')
     created_at_range = {"created_at": {"format": "epoch_millis"}}
+    change_ids = params.get('change_ids')
```
Comment (morucci) [inline monocle/db/queries.py:74]: Because the params is now handled there, that is ok.

File: monocle/db/queries.py [hunk 7]
```diff
@@ -476,11 +467,11 @@ def hot_changes(es, index, repository_fullname, params):
     _params = {
         'etype': ('Change',),
         'state': 'OPEN',
-        'repository_fullname_and_number': change_ids,
+        'change_id': change_ids,
```
Comment (morucci) [inline monocle/db/queries.py:470]: Could you change for "change_ids' as the params key ?

File: web/src/App.js (hunk 1 above)
Comment (flepied) [reply web/src/App.js:211]: I tried to remove it without success so far...

File: monocle/db/queries.py (hunk 2 above)
Comment (flepied) [reply monocle/db/queries.py:615]: ok changing it

File: web/src/App.js (hunk 1 above)
Comment (flepied) [reply web/src/App.js:211]: ok I got clearer ideas this morning and managed to do it !

File: monocle/db/queries.py (hunk 2 above)
Comment (flepied) [reply monocle/db/queries.py]: done

File: monocle/db/queries.py (hunk 3 above)
Comment (flepied) [reply monocle/db/queries.py:90]: I don't get it. I need this especially for all_changes.

File: monocle/db/queries.py (hunk 4 above)
Comment (flepied) [reply monocle/db/queries.py:62]: done

File: monocle/db/queries.py (hunk 5 above)
Comment (flepied) [reply monocle/db/queries.py:59]: done

File: monocle/db/queries.py (hunk 6 above)
Comment (flepied) [reply monocle/db/queries.py:74]: ok

File: monocle/db/queries.py (hunk 7 above)
Comment (flepied) [reply monocle/db/queries.py:470]: done

File: monocle/db/queries.py (hunk 3 above)
Comment (morucci) [reply monocle/db/queries.py:87]: The exec flow does not enter in generate_events_filter due to that condition.

Comment (flepied) [reply monocle/db/queries.py:87]: done

**Pull request merged**